site-src/geps/gep-718.md

```diff
@@ -0,0 +1,338 @@
+-	// precedence.
+-	//
+-	// If the referent cannot be found, the rule is not included in the route.
+-	// The controller should raise the "ResolvedRefs" condition on the Gateway
+-	// with the "DegradedRoutes" reason. The gateway status for this route should
+-	// be updated with a condition that describes the error more specifically.
++	// Group is the group of the backend resource.
+ 	//
+-	// Support: Custom
+-	//
+-	// +optional
+-	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
++	// +kubebuilder:validation:MinLength=1
++	// +kubebuilder:validation:MaxLength=253
++	Group string `json:"group"`
+ 
+ 	// Port specifies the destination port number to use for the
+-	// backend referenced by the ServiceName or BackendRef field.
+-	// If unspecified, the destination port in the request is used
+-	// when forwarding to a backendRef or serviceName.
++	// backend resource.
++	// This field is required when the backend is a Kubernetes Service.
+ 	//
+ 	// Support: Core
+ 	//
+@@ -146,11 +134,10 @@ type RouteForwardTo struct {
+ 	Port *PortNumber `json:"port,omitempty"`
+ 
+ 	// Weight specifies the proportion of HTTP requests forwarded to the backend
+-	// referenced by the ServiceName or BackendRef field. This is computed as
+-	// weight/(sum of all weights in this ForwardTo list). For non-zero values,
+-	// there may be some epsilon from the exact proportion defined here
+-	// depending on the precision an implementation supports. Weight is not a
+-	// percentage and the sum of weights does not need to equal 100.
++	// This is computed as weight/(sum of all weights in this Backends list).
++	// For non-zero values, there may be some epsilon from the exact proportion
++	// defined here depending on the precision an implementation supports. Weight
++	// is not a percentage and the sum of weights does not need to equal 100.
+ 	//
+ 	// If only one backend is specified and it has a weight greater than 0, 100%
+ 	// of the traffic is forwarded to that backend. If weight is set to 0, no
+```
+
+## Examples
+
+
+For Kubernetes Services, an updated `forwardTo` section will read as follows:
+
+```yaml
+...
+backends:
+- name: foo-service-v1
+  port: 80
+  weight: 80
+- name: foo-service-canary
+  port: 80
+  weight: 20
+...
+```
+
+Here, the `kind` field is omitted as it will be injected as a default.
+
+For custom backends, the API will look like the following:
+
+```yaml
+...
+backends:
+- name: foo-v1
+  kind: server
+  group: networking.acme.io
+  port: 80
+  weight: 80
+- name: foo-v1-canary
+  kind: server
+  group: networking.acme.io
+  port: 80
+  weight: 20
+...
+```
+
+For completeness, here is an example of HTTPRoute:
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: bar-route
+  labels:
+    gateway: prod-web-gw
+spec:
+  hostnames:
+  - "bar.example.com"
+  rules:
+  - matches:
+    - headers:
+        type: Exact
+        values:
+          env: canary
+    backends:
```

Comment:
Suggestions so far on this name:
- `backendRefs`: `refs` doesn't add much value here
- `targetRefs`: `target` feels a bit too generic here

Reply:
Destination and/or source is almost always preferable to "target". Target is ambiguous because it could be talking about the source's target (which is the destination) or the *rule's* target, (which is often the source).